### PR TITLE
[Widgets editor] Update the build process to account for legacy-widget block living in edit-widgets package

### PR DESCRIPTION
**Pull request by adamziel**
## Description

In #25371 we moved the legacy-widget block to edit-widgets package. This PR updates webpack pipelines to ensure we correctly handle the new block during the build.

One more follow-up will be required to ensure `wordpress-develop` handles this correctly.

Alternatively we could just move that block into block-library during the build, but I have some concerns about that - in particular it seems pretty counter-intuitive.

## How has this been tested?
1. Remove your existing build directory
1. Run `npm run dev`
1. Go to widgets editor
1. Confirm the legacy widget preview is broken
1. Kill `npm run dev`
1. Apply this PR
1. Run `npm run dev`
1. Go to widgets editor
1. Confirm the legacy widget preview is working

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -211,6 +213,21 @@ module.exports = {
...
+				to: 'build/edit-widgets/blocks/[1].php',
+				transform: transformBlockContent,
+			},
+			{
+				from: './packages/edit-widgets/src/blocks/*/block.json',
+				test: new RegExp(
+					`([\\w-]+)${ escapeRegExp( sep ) }block\\.json$`
+				),
+				to: 'build/edit-widgets/blocks/[1]/block.json',
+			},
```
Comment (youknowriad) [inline webpack.config.js:230]: Seems like we could simplify this a bit further by having an array of paths where block could live and generate this with a loop?

Comment (adamziel) [reply webpack.config.js:230]: Good call!

File: webpack.config.js [hunk 2]
```diff
@@ -41,6 +41,63 @@ const gutenbergPackages = Object.keys( dependencies )
...
+			// callback, the Gutenberg build blocks are modified
+			// to occur at a later priority.
+			.replace(
+				/(add_action\(\s*'init',\s*'gutenberg_register_block_[^']+'(?!,))/,
+				'$1, 20'
+			)
+	);
+};
+
+const blocksDirectoryMapping = toPairs( {
```
Comment (kevin940726) [inline webpack.config.js:78]: Nitpick: Personally, I'm not a fan of over-using lodash in simple cases like this one, prefer just a simple `Object.entries()` call 😅

File: webpack.config.js [hunk 3]
```diff
@@ -41,6 +41,63 @@ const gutenbergPackages = Object.keys( dependencies )
...
+	'./packages/block-library/src/': 'build/block-library/blocks/',
+	'./packages/edit-widgets/src/blocks/': 'build/edit-widgets/blocks/',
+} );
+
+const copyBlocksWebpackMapping = blocksDirectoryMapping.flatMap(
```
Comment (kevin940726) [inline webpack.config.js:83]: Nitpick: Since `blocksDirectoryMapping` is only used here, maybe just chain them together?

**Review by kevin940726 — APPROVED** [review]
Code LGTM 👍

**Review by draganescu — APPROVED** [review]
Tested and it works!

**Pull request merged**